hal/armv8m/spinlock.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * Spinlock
+ *
+ * Copyright 2017, 2022 Phoenix Systems
+ * Author: Pawel Pisarczyk, Damian Loewnau
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "../spinlock.h"
+
+struct {
+	spinlock_t spinlock;
+	spinlock_t *first;
+} spinlocks;
+
+
+void hal_spinlockSet(spinlock_t *spinlock, spinlock_ctx_t *sc)
+{
+	/* clang-format off */
+	__asm__ volatile(" \
+		mrs r1, primask; \
+		cpsid i; \
+		str r1, [%0]; \
+		mov r2, #0; \
+	1: \
+		ldrexb r1, [%1]; \
+		cmp r1, #0; \
+		beq 1b; \
+		strexb r1, r2, [%1]; \
+		cmp r1, #0; \
+		bne 1b; \
+		dmb"
+	:
+	: "r" (sc), "r" (&spinlock->lock)
+	: "r1", "r2", "memory", "cc");
```

Comment:
```suggestion
		mrs r2, primask; \
		cpsid i; \
		str r2, [%0]; \
		mov r3, #0; \
	1: \
		ldrexb r2, [%1]; \
		cmp r2, #0; \
		beq 1b; \
		strexb r2, r3, [%1]; \
		cmp r2, #0; \
		bne 1b; \
		dmb"
	:
	: "r" (sc), "r" (&spinlock->lock)
	: "r2", "r3", "memory", "cc");
```

Reply:
Why do we want to avoid using r1?

Reply:
Now I am thinking whether "cc" is even  needed in clobbers